aea/helpers/dialogue/base.py

```diff
@@ -656,6 +688,31 @@ def new_self_initiated_dialogue_reference(self) -> Tuple[str, str]:
         """
         return str(self._dialogue_nonce + 1), ""
 
+    def create(self, counterparty, performative, **kwargs,) -> Tuple[Message, Dialogue]:
+        initial_message = self._message_class(
+            dialogue_reference=("", ""),
```

Comment:
wrong; we have a constructor method for the dialogue reference which needs to be used here.

Reply:
When a message is being created in the context of a dialogue, its `dialogue_reference` is set to the dialogue's `dialogue_reference` (e.g. in `dialogue.reply(...)`).

But for an initial message here, the dialogue hasn't been created yet, and the dialogue cannot be created before the message because for it we need the message to set the role. That's why I create the initial message first, with some arbitrary dialogue_reference`, then I create an empty dialogue off of this message. Then I update the message's `dialogue_reference` to be the dialogue's, then add the message to the dialogue and return.

Reply:
That's incorrect. Look at skills usage; we always create the label before we create the dialogue...https://github.com/fetchai/agents-aea/blob/master/packages/fetchai/skills/generic_buyer/handlers.py#L423

Reply:
Done